packages/api/src/models/player.ts

```diff
@@ -73,16 +79,45 @@ export class PlayerModel {
 
   public async update(player: DbPlayer): Promise<DbPlayer> {
     const { username } = player
-    const exists = await this.repository.getOne({ username })
-
-    if (!exists) {
-      throw new Error(`Player does not exist (name: ${player.username})`)
-    }
+    // const exists = await this.repository.getOne({ username })
 
     return await this.repository.updateOne({ username }, player)
   }
 
   public async get(key: string): Promise<WithId<DbPlayer> | null> {
     return await this.repository.getOne({ key })
   }
+
+  public generateResource(
+    player: WithId<DbPlayer>,
+    lastTrade: WithId<DbTrade> | null
+  ): Resource {
+    // Compute points
+    let amount
+    if (!lastTrade) {
+      amount = TRADE_POINTS
+    } else {
+      amount = Math.max(
+        Math.ceil(lastTrade.resource.amount / TRADE_POINTS_DIVISOR),
+        TRADE_POINTS_MIN
+      )
+    }
+
+    // Get trait
+    const ranchName = RanchName[player.ranch]
+    const trait = ranchToTrait[ranchName]
+
+    return {
+      amount,
+      trait,
+    }
+  }
+
+  public async setTradeCooldown(from: string, to: string): Promise<boolean> {
+    return true
```

Comment:
I guess this is not implemented yet?

Reply:
This function is not being used. We calculate the cooldown asking for the last trade. I'll delete this